src/safeposix/syscalls/fs_calls.rs

```diff
@@ -389,42 +389,121 @@ impl Cage {
         }
     }
 
-    //------------------MKNOD SYSCALL------------------
-
+    /// ## ------------------MKNOD SYSCALL------------------
+    /// ### Description
+    ///
+    /// The `mknod_syscall()` creates a filesystem node (file, device special file
+    /// or pipe) named by a path as the input parameter.
+    /// The file type and the permissions of the new file are initialized from the
+    /// "mode" provided as the input parameter.
+    /// There are 5 different file types: S_IFREG, S_IFCHR, S_IFBLK, S_IFIFO, or
+    /// S_IFSOCK representing a regular file, character special file, block special
+    /// file, FIFO (named pipe), or UNIX domain socket, respectively.
+    /// The newly created file is empty with size 0.
+    /// On successful completion, the timestamps for both the newly created file
+    /// and its parent are updated along with their linkcounts.
+    ///
+    /// ### Function Arguments
+    ///
+    /// The `mknod_syscall()` receives three arguments:
+    /// * `path` - This argument points to a pathname naming the file.
+    /// For example: "/parentdir/file" represents the new file name as "file",
+    /// which will be created at this path (/parentdir/file).
+    ///
+    /// * `mode` - The mode argument specifies both the permissions to use and the
+    /// type of node to be created. It is a combination (using bitwise OR) of one
+    /// of the file types and the permissions for the new node.
+    /// FileType - In LIND, we have only implemented the file type of "Character
+    /// Device" represented by S_IFCHR flag.
+    /// FilePermission - The general permission mode used is "S_IRWXA": which
+    /// represents the read, write, and search permissions on the new file.
+    /// The final file mode is represented by the bitwise-OR of FileType and
+    /// FilePermission Flags.
+    ///
+    /// * `dev` - It is a configuration-dependent specification of a character or
+    /// block I/O device. If mode does not indicate a block special or character
+    /// special device, dev is ignored.
+    /// Since "CharDev" is the only supported type, 'dev' is represented using
+    /// makedev() function; that returns a formatted device number   
+    /// For example: "makedev(&DevNo { major: majorId, minor: minorId })" accepts a
+    /// Device Number that consists of a MajorID, identifying the class of the device,
+    /// and a minor ID, identifying a specific instance of a device in that class.
+    ///
+    /// ### Returns
+    ///
+    /// Upon successful creation of the file, 0 is returned.
+    /// Otherwise, errors or panics are returned for different scenarios.
+    ///
+    /// ### Errors
+    ///
+    /// * `ENOENT` - occurs when a directory component in the absolute path does
+    /// not exist
+    /// * `EPERM` - the mode bits for the new file are not sane
```

Comment:
What does it mean for the mode bits to not be sane?

Reply:
It means to check if the mode bits, which are passed as function arguments are correctly set or not. If the mode bits are correctly set, then we should be able to create a new file with required file permissions which are represented by the "mode" bits. Otherwise, we can't create a file and return an error.